src/AppChooser/Dialog.vala

```diff
@@ -159,22 +142,27 @@ public class AppChooser.Dialog : Hdy.Window {
         grid.attach (switcher, 0, 4, 2);
         grid.attach (button_box, 1, 5);
 
-        handle.add (grid);
-        add (handle);
+        var window_handle = new Hdy.WindowHandle ();
+        window_handle.add (grid);
+
+        add (window_handle);
+
+        if (parent_window != "") {
+            var parent = ExternalWindow.from_handle (parent_window);
+
+            if (parent == null) {
+                warning ("Failed to associate portal window with parent window %s", parent_window);
+            } else {
+                parent.set_parent_of (get_window ());
+            }
+        }
```

Comment:
this should be done in `realize ()` otherwise `get_window ()` return null.
```suggestion
        realize.connect (() => {
            if (parent_window != "") {
                var parent = ExternalWindow.from_handle (parent_window);

                if (parent == null) {
                    warning ("Failed to associate portal window with parent window %s", parent_window);
                } else {
                    parent.set_parent_of (get_window ());
                }
            }
        });
```